### PR TITLE
 Cleanup converters in ComponentModel

### PR DESCRIPTION
**Pull request by hughbe**
No functional changes here apart from dead code removal. No additional tests though to make this PR a bit smaller

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs [hunk 1]
```diff
@@ -50,8 +51,10 @@ public static int CreateMask(int previous)
             return previous == 0 ? 1 : previous << 1;
         }
 
+        [ExcludeFromCodeCoverage]
         public override bool Equals(object o) => o is InterlockedBitVector32 ? _data == ((InterlockedBitVector32)o)._data : false;
 
+        [ExcludeFromCodeCoverage]
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs]: Why did you add these attributes?

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/LicenseException.cs [hunk 2]
```diff
@@ -16,6 +16,8 @@ namespace System.ComponentModel
     [SuppressMessage("Microsoft.Design", "CA1032:ImplementStandardExceptionConstructors")] // must not, a Type is required in all constructors.
     public class LicenseException : SystemException
     {
+        private const int License = unchecked((int)0x80131901);
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/LicenseException.cs]: At least keep "HResult" in the name.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs (hunk 1 above)
Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs]: This was dead code and I wasn't sure if I should get rid of it though. The constructor was unused and can definitely be deleted.

I audited the code and the only use of `InterlockedBitVector32` is in https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectPropertyDescriptor.cs#L64

No uses of state call `Equals` or `GetHashCode`, so this is dead:
https://github.com/dotnet/corefx/blob/master/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectPropertyDescriptor.cs#L64

Comment (stephentoub) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs]: We shouldn't be adding such attributes, especially if it's dead code (how would someone know to look at these in the future for possible removal?).

I wouldn't remove the code, though.  structs like this are supposed to override GetHashCode and Equals.  If someone ended up using these in the future and we removed them, the caller would get the base implementation's that is really expensive.  I'd prefer to err on the side of just leaving them as-is.

Comment (hughbe) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs]: Okay, `GetHashCode` and `Equals` have returned! I do think that the ExcludeFromCodeCoverageAttribute is helpful here, so I kept it and added comments

Comment (stephentoub) [reply src/System.ComponentModel.TypeConverter/src/System/ComponentModel/InterlockedBitVector32.cs]: > I do think that the ExcludeFromCodeCoverageAttribute is helpful here,

Why? It's effectively a lie. We should only be using it for things that we don't care about (e.g. test assemblies) rather than trying to improve a vanity metric.

File: src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs [hunk 3]
```diff
@@ -2913,31 +2843,35 @@ internal FilterCacheItem(ITypeDescriptorFilterService filterService, ICollection
 
             internal bool IsValid(ITypeDescriptorFilterService filterService)
             {
-                if (!Object.ReferenceEquals(_filterService, filterService)) return false;
+                if (!ReferenceEquals(_filterService, filterService))
+                {
+                    return false;
+                }
+
```
Comment (stephentoub) [inline src/System.ComponentModel.TypeConverter/src/System/ComponentModel/TypeDescriptor.cs]: This can just be:
```C#
return ReferenceEquals(_filterService, filterService);
```

**Comment by stephentoub** [comment]
Thanks, @hughbe.
